pkg/util/resourceinformer/resourceinformer.go

```diff
@@ -0,0 +1,151 @@
+package resourceinformer
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	workv1alpha1 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/util"
+	"github.com/karmada-io/karmada/pkg/util/fedinformer/genericmanager"
+	"github.com/karmada-io/karmada/pkg/util/fedinformer/keys"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+	"github.com/karmada-io/karmada/pkg/util/restmapper"
+)
+
+var _ ResourceInformer = &resourceInformerImpl{}
+
+type ResourceInformer interface {
+	// BuildResourceInformers builds informer dynamically for managed resources in member cluster.
+	// The created informer watches resource change and then sync to the relevant Work object.
+	BuildResourceInformers(cluster *clusterv1alpha1.Cluster, work *workv1alpha1.Work, handler cache.ResourceEventHandler) (controllerruntime.Result, error)
```

Comment:
+1 

`controllerruntime.Result` doesn't seem to be a generic return type. I believe it would be beneficial to declare a new type for easier future expansion.

Reply:
Done.